frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.tsx

```diff
@@ -128,19 +134,22 @@ function EmbedFrame({
               />
             )}
             {hasParameters && (
-              <div className="flex">
+              <ParametersWidgetContainer>
                 <SyncedParametersList
                   className="mt1"
                   question={question}
                   dashboard={dashboard}
                   parameters={getValuePopulatedParameters(
                     parameters,
-                    parameterValues,
+                    _.isEmpty(draftParameterValues)
+                      ? parameterValues
+                      : draftParameterValues,
                   )}
                   setParameterValue={setParameterValue}
                   hideParameters={hide_parameters}
                 />
-              </div>
+                {dashboard && <FilterApplyButton />}
```

Comment:
Since `EmbedFrame` not only renders dashboards but questions as well. And in those cases `FilterApplyButton` will crash since it expects a dashboard to exist.